### ISSUE TITLE
Switch to Netlify

### ISSUE THREAD

**Issue by garbas** [NixOS/nixos-homepage#326]
Regardless what we will do with the website content and design it would be nice going forward to be able to iterate on the website faster as is currently the case.

A nice thing coming out of this migration is that we would have to cleanup our setup a bit and move certain things under subdomains (eg.

Now Netlify is quite popular provider for hosting static websites and integrated with Github, especially I'm interested in having previews for pull requests.

If we later decide to switch to some other provider for static websites (and CDN) I would discuss this once the migration to Netlify is done. Since migration of static site will be then much easier. For now lets not kill the energy behind the idea :)

I'm using this as a meta ticket to point to different PRs which I'm going to open as I will try to do this work.

**Comment by samueldr** [comment]
What is the scope of vendor lock-in, if any, in Netlify?

* * *

I'll add a note that, when NixOS proper moves, I'm perfectly fine to moving the `mobile.nixos.org` site to this service too.

**Comment by samueldr** [comment]
I see [they do analytics](https://www.netlify.com/products/analytics/). Is this anything we need to concern ourselves with legaleses and notifications? I see it's opt-in at a cost, but are they still collecting the data when not enabled?

**Comment by zimbatm** [comment]
This is just an upgrade that they offer. TBH most of the work here will be to re-shape this repo to make it buildable by Netlify. Once that happens a nice side-effect will be that it will become trivial to deploy to other targets as well.

**Comment by samueldr** [comment]
> This is just an upgrade [...]

Yes, but my concern are, maybe unfounded, about if they collect the information anyway.

**Comment by garbas** [comment]
Hi @samueldr! As @zimbatm already mentioned, moving to Netlify would force us to make a clear separation between our website and artifacts folders (nixos.org/releases). Regardless if we stay in the future with Netlify or not we will benefit from it. Analytics is an add-on which you need to enable. Do we need it? Maybe, but I would prefer this to be a separate discussion. 

It is also good to hear about moving mobile.nixos.org. You want me to ping you once we are done with nixos.org and maybe then it would be a simple copy/paste job?

**Comment by samueldr** [comment]
Sure, ping me once this is going on / done.

**Comment by samueldr** [comment]
:/ https://github.com/NixOS/nixos-homepage/commit/767a50a243eaa530435d2a3b89b41c8ca620144f#commitcomment-37725171

Those repeated publish/deploy comments are not great for those watching the repo.

**Comment by edolstra** [comment]
nixos.org points to Netlify now.

One remaining issue: currently it points to Netlify using an A record because you can't have CNAMEs on apex records. So IPv6 doesn't work and load balancing is probably also affected. Solutions: 1) Switch from nixos.org to www.nixos.org (https://www.netlify.com/blog/2017/02/28/to-www-or-not-www/); or 2) Use an ALIAS record on the apex, however united-domains doesn't support this so we'll have to switch to another DNS. We already have a terraform config for route53 but switching to route53 breaks the email forwarding provided by united-domains. So probably we should find some email forwarding service.

**Comment by edolstra** [comment]
What should we do about planet.nixos.org? It's currently hosted on the same server which will be shut down soon.

**Comment by samueldr** [comment]
@zimbatm re planet.nixos.org, could it be relevant to the [nix-community](https://github.com/nix-community/) org? Though that would only be shifting responsibilities.

**Comment by edolstra** [comment]
Created a separate issue for planet.nixos.org: #336.

**Comment by garbas** [comment]
I think we can close this one. we are still proxying install script but i think this is just going to stay *forever*.
